src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/SymbolKey/SymbolKey.NamedTypeSymbolKey.cs

```diff
@@ -71,12 +71,12 @@ public static SymbolKeyResolution Resolve(SymbolKeyReader reader, out string? fa
             private static void Resolve(
                 PooledArrayBuilder<INamedTypeSymbol> result,
                 INamespaceOrTypeSymbol container,
-                string metadataName,
+                string name,
                 int arity,
                 bool isUnboundGenericType,
                 ITypeSymbol[] typeArguments)
             {
-                foreach (var type in container.GetTypeMembers(GetName(metadataName), arity))
+                foreach (var type in container.GetTypeMembers(name, arity))
```

Comment:
we used the metadata name, only to break it apart here... i'm not sure what benefit that has at all.  so i removed this logic.  i can't see hwo this would break.  making a PR to see if we have any sort of issue here.